### PR TITLE
[Security][Encrypting] Lockbox & blind_index gems installation (1/5)

### PR DESCRIPTION
**Pull request by begault**
# Issues #8 & #9 

- Add of `lockbox` gem (Modern encryption for Rails)
- Add of `blind_index` gem (Securely search encrypted database fields)
- Encryption fields creation: 
   - User model: Emails
   - Rebels model: Emails / Phones

# References
https://ankane.org/sensitive-data-rails
https://ankane.org/securing-user-emails-lockbox
https://github.com/ankane/blind_index
https://github.com/ankane/lockbox

# Notes
- Before deploying this code, generate security key
`Lockbox.generate_key` 
- Save the security key with other secrets that should not be accessible to anybody except admin dev
- Expected environment variables format: `LOCKBOX_MASTER_KEY` (Currently lockbox & blind_index share the same master key, feel free to ask for separation (`BLIND_INDEX_MASTER_KEY`)) 
- After deploying this code, run the following command to migrate data: 
` Lockbox.migrate(User)`
`Lockbox.migrate(Rebel)`

# Next steps
- Ignore the legacy `email` field of `User` model (https://github.com/extinctionrebellion/RebelsManager/pull/93)
- Ignore the legacy `phone` and `email` fields of `Rebel` model (https://github.com/extinctionrebellion/RebelsManager/pull/93)
- Remove the legacy columns (https://github.com/extinctionrebellion/RebelsManager/pull/94)
- Remove the encrypted fields from logs (https://github.com/extinctionrebellion/RebelsManager/pull/95)
- Add encrypted reconfirm system ?

### CODE REVIEW

File: db/migrate/20191020155612_add_email_ciphertext_to_rebels.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class AddEmailCiphertextToRebels < ActiveRecord::Migration[5.2]
```
Comment (begault) [inline db/migrate/20191020155612_add_email_ciphertext_to_rebels.rb]: Migration version has to upgrade

**Comment by mhulet** [comment]
I have [added a migration](https://github.com/extinctionrebellion/RebelsManager/pull/92/commits/b98fca03ed515a8a1e1a2f08c364841cee4f61cf) to set `:phone_bidx` uniqueness to `false`. It was failing locally with empty phone numbers, setting the same value multiple times.

**Comment by mhulet** [comment]
Thanks a lot @begault, for the code and for the PR description 🙏

I have gone locally from step 1 to step 3 with a few issues, and have pushed my commits. I'll wait for your review before going further, and once we are all set we'll deploy each step.

File: db/migrate/20191020155822_add_phone_ciphertext_to_rebels.rb [hunk 2]
```diff
@@ -5,6 +5,6 @@ def change
 
     # blind index
     add_column :rebels, :phone_bidx, :string
-    add_index :rebels, :phone_bidx, unique: true
```
Comment (begault) [inline db/migrate/20191020155822_add_phone_ciphertext_to_rebels.rb]: @mhulet If you add a migration to update the index, there is no need to update my migration. 
The easiest would be to keep the `add_phone_ciphertext_to_rebels.rb` update and remove the new migration `update_rebels_phone_bidx_uniqueness.rb`

**Comment by begault** [comment]
@mhulet  
Thanks for the review and the migration update ! 
I've remove the duplicate migration and have added some updates about fields descriptions in models. I let you check again and update me about next actions.

**Comment by mhulet** [comment]
@begault This is now live, thanks so much for the documentation and the way the PR were organised 🙌👍

**Comment by begault** [comment]
Happy to see all of this live :) 
Thanks for your time answering to me & for the deployment process !

**Pull request merged**